install/kubernetes/cilium/templates/cilium-envoy/daemonset.yaml

```
@@ -86,7 +86,7 @@ spec:
         {{- if semverCompare ">=1.20-0" .Capabilities.KubeVersion.Version }}
         startupProbe:
           httpGet:
-            host: "localhost"
+            host: {{ .Values.ipv4.enabled | ternary "127.0.0.1" "::1" | quote }}
```

WDYT about the below instead? The main change is to use ::1 for dual stack.

```suggestion
            host: {{ .Values.ipv6.enabled | ternary "::1" "127.0.0.1" | quote }}
```

---

We are using the same expression in other places, let's keep your change for consistency.

---

I'm all for prioritising IPv6 over IPv4 where practical - in this case I had copied the logic from https://github.com/cilium/cilium/blob/cb108f247c97a4e26851b8613c920c830d793a00/install/kubernetes/cilium/files/cilium-envoy/configmap/bootstrap-config.json#L67

If we make the change here, I'd suggest making the same change in `bootstrap-config.json` too

---

yup, I found out about other place as well, so let's keep your change 👍